gcs/innards/proof.cc

```diff
@@ -1178,24 +1178,56 @@ auto Proof::emit_red_proof_line(const SumLessEqual<Weighted<PseudoBooleanTerm>>
     return record_proof_line(++_imp->proof_line, level);
 }
 
-auto Proof::emit_red_proof_lines_reifying(const SumLessEqual<Weighted<PseudoBooleanTerm>> & ineq, ProofFlag reif,
-    ProofLevel level) -> pair<ProofLine, ProofLine>
+auto Proof::emit_subproofs(const map<string, JustifyExplicitly> & subproofs)
 {
-    need_all_proof_names_in(ineq.lhs);
+    _imp->proof << " begin\n";
+    ++_imp->proof_line;
+    for (const auto & [proofgoal, proof] : subproofs) {
+        ++_imp->proof_line;
```

Comment:
This looks fragile and prone to going horribly wrong. Perhaps it's better to pass in a lambda function that gets called, and deal with constraint IDs that way? It's probably useful for the subproof-writing code to have access to constraint IDs.

Reply:
Going to leave these for now since it will need rewritten anyway with the new proof API?